docs: add note about Consul Enterprise role bindings (#25426)
hashicorp/nomad: website/content/docs/configuration/consul.mdx

  namespace](/consul/docs/enterprise/namespaces) used by the Consul
  integration. If non-empty, this namespace will be used on all Consul API calls
  and for Consul Connect configurations, unless overridden by the job's
  [`consul.namespace`][] field. In Nomad Community Edition, only the `"default"`

>   namespace will be used, so this field should be omitted.

Reviewer: ```suggestion
  namespace is used, so you should omit this field.
```
